cogctl/cli/main.py

```diff
@@ -50,39 +54,38 @@ def cli(ctx, config_file, profile, url, user, password, verbose):
     """
 
     state = State()
-
     config_file = os.path.abspath(os.path.expanduser(config_file))
-    state.config_file = config_file
     state.verbosity = verbose
 
-    if os.path.isfile(config_file):
-        try:
-            state.configuration = config.read_config(config_file)
-        except KeyError:
-            raise click.BadParameter("The configuration file '%s' does not "
-                                     "specify a default profile, and is thus "
-                                     "invalid" % config_file,
-                                     param_hint=['--config-file'])
+    try:
+        state.configuration = CogctlConfig(config_file)
+    except KeyError:
+        raise click.BadParameter("The given configuration file is not valid",
+                                 param_hint=["--config-file"])
 
     if profile:
-        if state.configuration:
-            profile_data = state.configuration.get(profile)
-            if profile_data:
-                state.profile = state.configuration[profile]
-            else:
-                raise click.BadParameter("Profile '%s' was not found in "
-                                         "configuration file '%s'"
-                                         % (profile, config_file),
-                                         param_hint=['--profile'])
-        else:
-            raise click.BadParameter("If you specify a profile, the "
-                                     "configuration file must exist",
-                                     param_hint=['--config-file', '--profile'])
+        try:
+            state.profile = state.configuration.profile(profile)
+        except KeyError:
+            raise click.BadParameter("Profile '{}' was not found in "
+                                     "configuration file '{}'".format(
+                                         profile, config_file),
+                                     param_hint=['--profile'])
     else:
-        if state.configuration:
-            profile = next((p for p in state.configuration.values()
-                            if p['default']))
-            state.profile = profile
+        try:
+            state.profile = state.configuration.default()
+        except KeyError:
+            # If we go this far, if the file exists, it already has a
+            # default profile. Thus, this only happens when the config
+            # file does not exist, which can be OK when a user is
+            # first setting up.
+
+            # TODO: test behavior when no cogctl file is present, and
+            # sufficient profile information is not present, but user
+            # attempts to make an API call
+            #
```

Comment:
Yeah, it would be easier. I was thinking that the prompt might be a nicer user experience, but it would kinda open up some potential funkiness when scripting with cogctl.

I think you're right, it's probably better to just fail.